src/Shared/Recruit.Vacancies.Client/Application/EventHandlers/NotifyOnVacancyActionsEventHandler.cs

```diff
@@ -5,41 +5,54 @@
 using Microsoft.Extensions.Logging;
 using System.Threading;
 using System.Threading.Tasks;
+using Communication.Types;
+using Esfa.Recruit.Vacancies.Client.Application.Communications;
 using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
 using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.StorageQueue;
 
 namespace Esfa.Recruit.Vacancies.Client.Application.EventHandlers
 {
     public class NotifyOnVacancyActionsEventHandler : INotificationHandler<VacancyClosedEvent>,
                                                         INotificationHandler<LiveVacancyUpdatedEvent>
     {
         private readonly INotifyVacancyUpdates _vacancyStatusNotifier;
-        private readonly IVacancyReviewRepository _vacancyReviewRepository;
         private readonly IVacancyRepository _vacancyRepository;
         private readonly ILogger<NotifyOnVacancyActionsEventHandler> _logger;
+        private readonly ICommunicationQueueService _communicationQueueService;
 
-        public NotifyOnVacancyActionsEventHandler(ILogger<NotifyOnVacancyActionsEventHandler> logger, INotifyVacancyUpdates vacancyStatusNotifier, IVacancyRepository vacancyRepository, IVacancyReviewRepository vacancyReviewRepository)
+        public NotifyOnVacancyActionsEventHandler(
+            ILogger<NotifyOnVacancyActionsEventHandler> logger, 
+            INotifyVacancyUpdates vacancyStatusNotifier, 
+            IVacancyRepository vacancyRepository, 
+            ICommunicationQueueService communicationQueueService)
         {
             _vacancyStatusNotifier = vacancyStatusNotifier;
-            _vacancyReviewRepository = vacancyReviewRepository;
             _vacancyRepository = vacancyRepository;
             _logger = logger;
+            _communicationQueueService = communicationQueueService;
         }
 
         public async Task Handle(VacancyClosedEvent notification, CancellationToken cancellationToken)
         {
-            try
+            var vacancy = await _vacancyRepository.GetVacancyAsync(notification.VacancyId);
+
+            if (vacancy.ClosureReason == ClosureReason.WithdrawnByQa)
```

Comment:
Should this go inside a try/catch like the Manual closure below?

Reply:
Try catch should be close to where we make the actual infrastructure call rather than interim classes unless we know how to treat the error situation.